### ISSUE TITLE
Dynamic color mapping for "fail-on" messages not working when using more than one reporter

### ISSUE THREAD

**Issue by felixp98** [PyCQA/pylint#10825]
### Bug description

In Issue https://github.com/pylint-dev/pylint/issues/9898 I added support for dynamic color mapping for messages/categories that are configured as "fail-on". The feature was shipped in pylint 4.0.0.

Unfortunately the implementation only works when using the colorized reporter on its own `--output-format=colorized`, but not when used with another output format combined `--output-format=json:out.json,colorized`.

I already got the pull request ready. I'm creating this issue only as reference for the towncrier changelog.

### Configuration

```ini
fail-on = ["missing-function-docstring"]
```

### Command used

```shell
pylint <package_dir> --output-format=json:out.json,colorized
```

### Pylint output

messages configured with fail-on are not colorized at all

### Expected behavior

messages configured with fail-on should be colored red inverse

### Pylint version

```shell
pylint 4.0.4
```

### Additional dependencies

```python

```
